apps/wallet/src/ui/app/components/accounts/EditNickname.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useZodForm } from '@mysten/core';
+import toast from 'react-hot-toast';
+import { useNavigate, useParams } from 'react-router-dom';
+import { z } from 'zod';
+import { useAccounts } from '../../hooks/useAccounts';
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { Button } from '../../shared/ButtonUI';
+import { Form } from '../../shared/forms/Form';
+import { TextField } from '../../shared/forms/TextField';
+import {
+	Dialog,
+	DialogContent,
+	DialogHeader,
+	DialogTitle,
+	DialogDescription,
+} from '_src/ui/app/shared/Dialog';
+
+const formSchema = z.object({
+	nickname: z.string().trim().nonempty('Required'),
+});
+
+export function EditNickname() {
+	const { accountID } = useParams();
+	const navigate = useNavigate();
+	const backgroundClient = useBackgroundClient();
+	const { data: accounts } = useAccounts();
+	const account = accounts?.find((account) => account.id === accountID);
+
+	const form = useZodForm({
+		mode: 'all',
+		schema: formSchema,
+		defaultValues: {
+			nickname: account?.nickname ?? '',
+		},
+	});
+	const {
+		register,
+		formState: { isSubmitting, isValid },
+	} = form;
+
+	const close = () => navigate('/accounts/manage');
+	const onSubmit = ({ nickname }: { nickname: string }) => {
+		if (account && accountID) {
+			try {
```

Comment:
try catch here wont work - `setAccountNickname` returns a promise.
I would probably change the onSubmit to be async and await it and the run close()
remove the unnecessary returns